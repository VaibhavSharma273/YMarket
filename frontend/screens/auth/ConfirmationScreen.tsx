import React, { useState } from 'react';
import { StyleSheet } from 'react-native';

import EditScreenInfo from '../../components/EditScreenInfo';
import { Text, View } from '../../components/Themed';
import TextInput from '../../components/TextInput';
import BackButton from '../../components/BackButton';
import { TouchableOpacity } from 'react-native';
// import { RootStackParamList, RootTabScreenProps } from '../types';
// import { StackScreenProps } from '@react-navigation/stack';

import { theme } from '../../assets/theme';
import { nameValidator } from '../../helpers/nameValidator';
import { phoneValidator } from '../../helpers/phoneValidator';

export default function RegisterScreen({ navigation }: any) {
    const [firstName, setFirstName] = useState({ value: '', error: '' });
    const [lastName, setLastName] = useState({ value: '', error: '' });
    const [phone, setPhone] = useState({ value: '', error: '' });

    const onSignUpPressed = () => {
        navigation.reset({
            index: 0,
            routes: [{ name: 'LoginScreen' }],
        });
    };

    return (
        <View style={styles.container}>
            <BackButton goBack={navigation.goBack} />
            {/* <View style={styles.separator} /> */}
            <TouchableOpacity style={styles.button} onPress={onSignUpPressed}>
                <Text style={styles.title}>Confirm Email</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
        color: 'white',
        fontFamily: 'Arial',
    },
    separator: {
        marginVertical: 10,
        height: 3,
        width: '50%',
    },
    header: {
        fontSize: 21,
        color: '#0f4d92',
        fontWeight: 'bold',
        paddingVertical: 12,
    },
    button: {
        alignItems: 'center',
        backgroundColor: '#0f4d92',
        padding: 10,
        marginTop: 20,
        width: 100,
    },
});
