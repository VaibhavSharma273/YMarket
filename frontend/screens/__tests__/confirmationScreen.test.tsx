import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import ConfirmationScreen from '../auth/ConfirmationScreen'
import { createTestProps } from '../../jest.setup'
import renderer from 'react-test-renderer';

describe('Confirmation Screen component', () => {
    let component: ShallowWrapper;
    let props: any;
    beforeEach(() => {
        props = createTestProps({});
        component = shallow(<ConfirmationScreen {...props}/>);
    })
    it(`renders correctly`, () => {
        expect(component.length).toBe(1)
    });
    it('matches the snapshot', () => {
        const tree = renderer.create(<ConfirmationScreen {...props}/>)
                             .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

