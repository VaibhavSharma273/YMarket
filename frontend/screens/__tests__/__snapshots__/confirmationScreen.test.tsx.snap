// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Screen component matches the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 4,
        "opacity": 1,
        "position": "absolute",
        "top": 30,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0f4d92",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "white",
            "fontFamily": "Arial",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Confirm Email
    </Text>
  </View>
</View>
`;
